lib/modules/dosomething/dosomething_api/resources/kudos_resource.inc

```diff
@@ -145,9 +148,17 @@ function _kudos_resource_access($op) {
   }
 
   if ($op === 'create' || $op === 'delete') {
-    if (!user_is_anonymous()) {
-      return TRUE;
+    // Only auth'd users should be giving Kudos
+    if (user_is_anonymous()) {
+      return FALSE;
+    }
+
+    // We need to make sure people are only creating Kudos for themselves
+    if ($op === 'create' && $user->uid !== $args[1]) {
```

Comment:
Yeah, I couldn't find any better solutions either when doing this for the magic link endpoints. 😞 Would it make sense to remove the `uid` value from the URL altogether since it always has to be the same as the logged-in user?